crypto/x509/x_ietfatt.c

```diff
@@ -0,0 +1,156 @@
+    STACK_OF(IETF_ATTR_SYNTAX_VALUE) *values;
+};
+
+ASN1_CHOICE(IETF_ATTR_SYNTAX_VALUE) = {
+    ASN1_SIMPLE(IETF_ATTR_SYNTAX_VALUE, u.octets, ASN1_OCTET_STRING),
+    ASN1_SIMPLE(IETF_ATTR_SYNTAX_VALUE, u.oid, ASN1_OBJECT),
+    ASN1_SIMPLE(IETF_ATTR_SYNTAX_VALUE, u.string, ASN1_UTF8STRING),
+} ASN1_CHOICE_END(IETF_ATTR_SYNTAX_VALUE)
+
+ASN1_SEQUENCE(IETF_ATTR_SYNTAX) = {
+    ASN1_IMP_SEQUENCE_OF_OPT(IETF_ATTR_SYNTAX, policyAuthority, GENERAL_NAME, 0),
+    ASN1_SEQUENCE_OF(IETF_ATTR_SYNTAX, values, IETF_ATTR_SYNTAX_VALUE),
+} ASN1_SEQUENCE_END(IETF_ATTR_SYNTAX)
+
+IMPLEMENT_ASN1_FUNCTIONS(IETF_ATTR_SYNTAX)
+IMPLEMENT_ASN1_FUNCTIONS(IETF_ATTR_SYNTAX_VALUE)
+
+int IETF_ATTR_SYNTAX_get_value_num(const IETF_ATTR_SYNTAX *a)
+{
+    return sk_IETF_ATTR_SYNTAX_VALUE_num(a->values);
+}
+
+const GENERAL_NAMES *
+IETF_ATTR_SYNTAX_get0_policyAuthority(const IETF_ATTR_SYNTAX *a)
+{
+    return a->policyAuthority;
+}
+
+void IETF_ATTR_SYNTAX_set0_policyAuthority(IETF_ATTR_SYNTAX *a,
+                                           GENERAL_NAMES *names)
+{
+    GENERAL_NAMES_free(a->policyAuthority);
+    a->policyAuthority = names;
+}
+
+void *IETF_ATTR_SYNTAX_get0_value(const IETF_ATTR_SYNTAX *a, int ind, int *type)
+{
+    IETF_ATTR_SYNTAX_VALUE *val;
+
+    val = sk_IETF_ATTR_SYNTAX_VALUE_value(a->values, ind);
+    if (type != NULL)
+        *type = val->type;
+
+    switch (val->type) {
+    case IETFAS_OCTETS:
+        return val->u.octets;
+    case IETFAS_OID:
+        return val->u.oid;
+    case IETFAS_STRING:
+        return val->u.string;
+    }
+
+    return NULL;
+}
+
+int IETF_ATTR_SYNTAX_add1_value(IETF_ATTR_SYNTAX *a, int type, void *data)
+{
+    IETF_ATTR_SYNTAX_VALUE *val;
+
+    if (data == NULL)
+        return 0;
+
+    if (a->values == NULL
+            && (a->values = sk_IETF_ATTR_SYNTAX_VALUE_new_null()) == NULL) {
+        ERR_raise(ERR_LIB_X509V3, ERR_R_MALLOC_FAILURE);
+        return 0;
+    }
+
+    val = IETF_ATTR_SYNTAX_VALUE_new();
+
+    val->type = type;
+    switch (type) {
+    case IETFAS_OCTETS:
+        val->u.octets = data;
+        break;
+    case IETFAS_OID:
+        val->u.oid = data;
+        break;
+    case IETFAS_STRING:
+        val->u.string = data;
+        break;
+    }
+
+    if (sk_IETF_ATTR_SYNTAX_VALUE_push(a->values, val) <= 0)
+        return 0;
+
+    return 1;
+}
+
+int IETF_ATTR_SYNTAX_print_ex(BIO *bp, IETF_ATTR_SYNTAX *a, int indent)
+{
+    if (a->policyAuthority != NULL) {
+        for (int i = 0; i < sk_GENERAL_NAME_num(a->policyAuthority); i++) {
+            if (BIO_printf(bp, "%*s", indent, "") <= 0)
+                goto err;
+
+            if (GENERAL_NAME_print(bp, sk_GENERAL_NAME_value(a->policyAuthority,
+                                                             i)) <= 0)
+                goto err;
+
+            if (BIO_printf(bp, "\n") <= 0)
+                goto err;
+        }
+    }
+
+    for (int i = 0; i < IETF_ATTR_SYNTAX_get_value_num(a); i++) {
```

Comment:
nit : `for (int i = 0; ...  ` is not compliant  with old _C89_ standard (not sure about C<number)>.
This counter variable declaration should be moved to the beginning of the method.